docs/book/src/topics/suspend-asg-processes.md

```diff
@@ -0,0 +1,82 @@
+# Suspend ASG Processes
+
+- **Feature status:** Experimental
+- **Feature gate:** MachinePool=true
+
+MachinePool allows users to manage many machines as a single entity. Infrastructure providers implement a separate CRD that handles infrastructure side of the feature.
+
+## Suspend Processes
+
+It's possible to suspend certain processes for ASG. The list of processes can be found [here](https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_SuspendProcesses.html).
+
+To utilize this feature, simply denote the list of processes that are desired to be suspended.
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSMachinePool
+metadata:
+  name: capa-mp-0
+spec:
+  minSize: 1
+  maxSize: 10
+  availabilityZones:
+    - "${AWS_AVAILABILITY_ZONE}"
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"  
+  suspendProcesses:
+    launch: true
+    alarmNotification: true
+    azRebalance: true
+---
+```
+
+## Resume Processes
+
+If a service is desired to be resumed, simply remove it from the list of suspended processes. The reconciler will then
+resume any process that is not part of the desired suspended processes list.
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSMachinePool
+metadata:
+  name: capa-mp-0
+spec:
+  minSize: 1
+  maxSize: 10
+  availabilityZones:
+    - "${AWS_AVAILABILITY_ZONE}"
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"  
+  suspendProcesses:
+    launch: true
+---
+```
+
+_Note_ that now `AlarmNotification` and `AZRebalance` will be resumed, but the reconciler will not try to suspend
+`Launch` again. So it doesn't incur additional expensive, redundant API calls.
+
+## Optional `All`
+
+An option is also provided to suspend all processes without having to set each of them to `true`. Simply use `all` like
+this:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSMachinePool
+metadata:
+  name: capa-mp-0
+spec:
+  minSize: 1
+  maxSize: 10
+  availabilityZones:
+    - "${AWS_AVAILABILITY_ZONE}"
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"  
+  suspendProcesses:
```

Comment:
@richardcase Ah, but I can't do that because all the fields will be set to false. I don't have a way to find out which was actually defined by a user or just left false. :/

Reply:
I guess I could make all of them pointers....

Reply:
@richardcase Are you okay with that? If I make all of them pointers so I can see if it was set at all? And if yes, I won't add that as a process.

Reply:
Originally i had thought about the `all` being at a different level:

```yaml
suspendProcesses:
   all: true
```

and 

```yaml
suspendProcesses:
  processes:
     azrebalance: true
     launch: false
     .....
```

Where the `processes` can be optional and a pointer....with webhooks to reject anything where `all` is true and `processes` is defined.

Reply:
Whatever you think is best @Skarlso :smile:

Reply:
Nah, I like the idea. It would work similarly to golangci-lint. Where you can enable all and then disable selected or disable all and enable selected. :)

Reply:
Added. :) With test. :)